pkg/virt-controller/ipamclaims/libipam/libipam_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package libipam_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/ipamclaims/libipam"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+const (
+	networkName = "network"
+)
+
+var _ = Describe("GetPersistentIPsConf", func() {
+	type testCase struct {
+		config           string
+		expectedAllowIPs bool
+		expectedName     string
+	}
+
+	DescribeTable("should return expected netConf",
+		func(tc testCase) {
+			nad := &networkv1.NetworkAttachmentDefinition{
+				Spec: networkv1.NetworkAttachmentDefinitionSpec{
+					Config: fmt.Sprintf(tc.config, networkName),
+				},
+			}
+			conf, err := libipam.GetPersistentIPsConf(nad)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(conf.AllowPersistentIPs).To(Equal(tc.expectedAllowIPs))
+			Expect(conf.Name).To(Equal(tc.expectedName))
```

Comment:
nit: you can inline all these

```suggestion
Expect(libipam.GetPersistentIPsConf(nad)).To(Equal(netConf{
	AllowPersistentIPs: tc.expectedAllowIPs,
	Name: tc.expectedName,
}))
```

Reply:
nice but can't because netConf isn't exported and i dont think we should export it for this
(unit test isn't the same package as the libipam production code so it can't use it)

Reply:
OK, good enough to keep it as is then.

Can you point me to any documentation why it is preferable to have the tests in a different pkg ? IIUC that just enforces you only test exported functions.

Reply:
It is what sig-network recommends,
I think it allows to decouple between the production code and the unit tests,
focusing only on the exported functions, which acts as API for the package
we can see that where we still don't adhere to that, we still test private functions, and then the tests
are sensitive to the internal implementation

https://softwareengineering.stackexchange.com/questions/290714/managing-code-unit-tests-with-source-or-separate
As Uncle Bob always says, your tests should not be coupled to the structure of your application ([great blog discussing this topic](http://blog.cleancoder.com/uncle-bob/2017/03/03/TDD-Harms-Architecture.html)).

I believe Edy would have lots to tell about it if desireable